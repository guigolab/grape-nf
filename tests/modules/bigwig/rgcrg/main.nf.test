nextflow_process {

    name "Test Process bigwig"
    script "modules/bigwig/rgcrg/main.nf"
    process "bigwig"

    tag "module"

    setup {
        run("fastaIndex") {
            script = "modules/fastaIndex/samtools/main.nf"
            process {
                """
                input[0] = "${baseDir}/data/genome.fa.gz"
                """
            }
        }
    }

    test("Should produce bigwig files using RGCRG tools") {

        when {
            params {
                // define parameters here. Example:
                // outdir = "tests/results"
            }
            process {
                """
                input[0] = fastaIndex.out
                input[1] = Channel.from([
                    [ "sample3", "test3", file("${baseDir}/data/sample3_m4_n10_toGenome.bam"), "bam", "GenomeAlignments", true, "MATE2_SENSE"]
                ])
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
