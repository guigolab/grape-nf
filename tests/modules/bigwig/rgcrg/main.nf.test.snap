{
    "Should produce bigwig files using RGCRG tools": {
        "content": [
            {
                "0": [
                    [
                        "sample3",
                        "test3",
                        [
                            "sample3.minusRaw.bw:md5,db10a50bb7eebec0f3421f4460d4cae8",
                            "sample3.plusRaw.bw:md5,235e2d65c856c969048e432234c04ff4"
                        ],
                        "bigWig",
                        [
                            "MinusRawSignal",
                            "PlusRawSignal"
                        ],
                        true,
                        "MATE2_SENSE"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-04-30T19:01:02.919082"
    }
}